src/push.rs

```diff
@@ -118,9 +164,7 @@ pub async fn push_profile(data: PushProfileData<'_>) -> Result<(), PushProfileEr
         (false, true) => build_command.arg("--no-link"),
     };
 
-    for extra_arg in data.extra_build_args {
-        build_command.arg(extra_arg);
-    }
+    build_command.args(data.extra_build_args);
 
     let build_exit_status = build_command
         // Logging should be in stderr, this just stops the store path from printing for no reason
```

Comment:
When I try this out and run it on my machine, it fails on line 181

```rust
  if !Path::new(
        format!(
            "{}/deploy-rs-activate",
            data.deploy_data.profile.profile_settings.path
        )
        .as_str(),
```

because, as the system is remote and the closure was built remotely, this path is not going to exist in my store or on my system.

I'm not sure I have a _great_ solution for that, but I wanted to point that out cause I'm trying this out in anticipation of managing remote machines via macOS

Reply:
Perhaps, in this case, activation should happen on the target as well :thinking:

Reply:
FTR, I have the same issue even with NixOS -> NixOS deployment :see_no_evil:

Reply:
How I ended up "sorta solving this" myself was to take the path that deploy info spits out and then dead-reckoning run a little bash function with that path as the input.

```sh
function activate-on-host(){                                                            
  printf -v profile %q "$1"                                                          
  ssh user@FQDN "export PROFILE=$profile ; bash $profile/deploy-rs-activate" 
}                                                                                    
```

Which, indeed, simply runs the activation on the target

Reply:
I see, I did not encounter that error, very likely because I had it built locally before I tested remote building

the push_profile function does quite a lote at this point and most of it assumes the derivation is being built locally
splitting it up into a function to handle building locally and another to handle remote builds should resolve some of those difficulties

Reply:
besides, contrary to its name it also handles the building, not just the pushing :thinking: